app/livechat/lib/LivechatInquiry.js

```diff
@@ -19,13 +19,31 @@ if (Meteor.isServer) {
 			this.tryEnsureIndex({ message: 1 }); // message sent by the client
 			this.tryEnsureIndex({ ts: 1 }); // timestamp
 			this.tryEnsureIndex({ agents: 1 }); // Id's of the agents who can see the inquiry (handle departments)
-			this.tryEnsureIndex({ status: 1 }); // 'open', 'taken'
+			this.tryEnsureIndex({ status: 1 }); // 'ready', 'queued', 'taken'
 		}
 
 		findOneById(inquiryId) {
 			return this.findOne({ _id: inquiryId });
 		}
 
+		findOneByRoomId(rid) {
```

Comment:
Every model method should expect a last parameter called `options` passing to the collection method allowing callers to filter fields, sort or so on.

Reply:
Yeah. Even the `findOne...` methods are more limited in terms of options, I do agree to apply this approach.
The point is that there are a lot of methods that do not respect this approach over the codebase.